### PR TITLE
ddl: skip the error when fallback from ingest to txn method

### PR DESCRIPTION
**Pull request by Benjamin2037**
### What problem does this PR solve?

Issue Number: ref #39321

Problem Summary:
When set fast reorg variable on, some ut will failed.

### What is changed and how it works?

> TestCancel
> TestMultiSchemaChangeAddIndexesCancelled
> TestCancelAddIndexPanic
> TestDDLStatsInfo
> TestPartitionAddIndexGC

This four case is dur to pd create rollback DDL, we set err to nil and DDL task will be resumed and executed later

> TestRunDDLJobPanicEnableClusteredIndex
> TestRunDDLJobPanicDisableClusteredIndex

skip merge woker number checker.

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: ddl/index.go [hunk 1]
```diff
@@ -823,7 +823,7 @@ func doReorgWorkForCreateIndex(w *worker, d *ddlCtx, t *meta.Meta, job *model.Jo
 					tryFallbackToTxnMerge(job, err)
 				}
 				ingest.LitBackCtxMgr.Unregister(job.ID)
-				return false, ver, errors.Trace(err)
+				return false, ver, nil
```
Comment (tangenta) [inline ddl/index.go:826]: If `convertAddIdxJob2RollbackJob` returns an error, it will be incorrectly hidden.

File: ddl/backfilling.go [hunk 2]
```diff
@@ -871,7 +871,10 @@ func (dc *ddlCtx) writePhysicalTableRecord(sessPool *sessionPool, t table.Physic
 	return nil
 }
 
-func injectCheckBackfillWorkerNum(curWorkerSize int) error {
+func injectCheckBackfillWorkerNum(curWorkerSize int, isMergeWorker bool) error {
+	if isMergeWorker {
+		return nil
+	}
```
Comment (tangenta) [inline ddl/backfilling.go:877]: Why is it skipped for the merge worker?

Comment (Benjamin2037) [reply ddl/backfilling.go:877]: Because merge index always be 1 in ut，ft.

File: ddl/index.go (hunk 1 above)
Comment (Benjamin2037) [reply ddl/index.go:826]:
```suggestion
				return false, ver, errors.Trace(err)
```

**Review by tangenta — APPROVED** [review]

File: ddl/index.go [hunk 3]
```diff
@@ -802,13 +802,13 @@ func doReorgWorkForCreateIndex(w *worker, d *ddlCtx, t *meta.Meta, job *model.Jo
...
 				tryFallbackToTxnMerge(job, err)
-				return false, ver, errors.Trace(err)
+				return false, ver, nil
 			}
 			done, ver, err = runReorgJobAndHandleErr(w, d, t, job, tbl, indexInfo, false)
 			if err != nil {
 				ingest.LitBackCtxMgr.Unregister(job.ID)
 				tryFallbackToTxnMerge(job, err)
-				return false, ver, errors.Trace(err)
+				return false, ver, nil
```
Comment (Benjamin2037) [inline ddl/index.go:811]:
```suggestion
				return false, ver, errors.Trace(err)
```

File: ddl/index.go [hunk 4]
```diff
@@ -753,12 +753,13 @@ func IngestJobsNotExisted(ctx sessionctx.Context) bool {
 }
 
 // tryFallbackToTxnMerge changes the reorg type to txn-merge if the lightning backfill meets something wrong.
-func tryFallbackToTxnMerge(job *model.Job, err error) {
+func tryFallbackToTxnMerge(job *model.Job, err *error) {
```
Comment (tangenta) [inline ddl/index.go:756]: Returning an error instead is much idiomatic.

**Review by hawkingrei — APPROVED** [review]

**Review by tangenta — APPROVED** [review]

**Pull request merged**